python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -45,7 +45,7 @@ def unique_and_compact_per_type(nodes):
             nodes, empty_tensor, empty_tensor
         )
         compacted = compacted.split(nums)
-        return unique, compacted
+        return unique, list(compacted)
```

Comment:
Why this was not caught by unit test?

Reply:
Type check is missed, UT just loop the output, while tuple is also iterable. Adding the missing part in the UT.